cli/src/main.rs

```diff
@@ -512,6 +519,102 @@ fn do_session(_: Context, c: Session) -> Result<String>
     Ok(base64::encode_config(&sid.0, base64::URL_SAFE_NO_PAD))
 }
 
+#[derive(Parser)]
+struct VerifyToken
+{
+    #[clap(subcommand)]
+    subcmd: VerifyTokenCmd,
+    #[clap(flatten)]
+    opts: VerifyTokenOpts,
+}
+
+#[derive(Parser)]
+struct VerifyTokenOpts
+{
+    #[clap(long, value_name = "endpoint", display_order = 1)]
+    url: Option<String>,
+}
+
+#[derive(Parser)]
+enum VerifyTokenCmd
+{
+    #[clap(display_order = 1)]
+    Create(VerifyTokenCreate),
+
+    #[clap(display_order = 2)]
+    Verify(VerifyTokenVerify),
+}
+
+fn do_verify_token(ctx: Context, h: VerifyToken) -> Result<String>
+{
+    match h.subcmd {
+        VerifyTokenCmd::Create(t) => do_verify_token_create(ctx, h.opts, t),
+        VerifyTokenCmd::Verify(t) => do_verify_token_verify(ctx, h.opts, t),
+    }
+}
+
+#[derive(Parser)]
+struct VerifyTokenCreate
+{
+    #[clap(long, value_name = "email", display_order = 1)]
+    email: Option<String>,
+
+    /// Identity seed to use. If specified, supersedes the configured value.
+    #[clap(long, value_name = "b64", display_order = 1)]
+    seed: Option<String>,
+}
+
+fn do_verify_token_create(
+    ctx: Context,
+    opt: VerifyTokenOpts,
+    c: VerifyTokenCreate,
+) -> Result<String>
+{
+    let cfg = load_conf(&ctx)?;
+    let url = opt.url.as_ref().unwrap_or(&cfg.api_host);
+
+    let id = match c.seed {
+        Some(s) => id_from_b64(s)?,
+        None => cli::load_seed(&cfg)?,
+    };
+
+    let v = cli::create_verify_token(url, id, c.email)
+        .context("cannot create verify token")?;
+
+    Ok(v)
+}
+
+#[derive(Parser)]
+struct VerifyTokenVerify
+{
+    #[clap(long, value_name = "secret", display_order = 1)]
+    secret: String,
+
+    /// Identity seed to use. If specified, supersedes the configured value.
+    #[clap(long, value_name = "b64", display_order = 2)]
+    seed: Option<String>,
```

Comment:
since the seed option here is shared between both create and verify subcommands, it probably should be placed on the verifytokenopts struct that's flattened into the umbrella verify subcommand